functional-test/src/main/java/org/zanata/page/WebDriverFactory.java

```diff
@@ -143,17 +141,18 @@ public void updateListenerTestName(String testName) {
         if (eventListener == null && ScreenshotDirForTest.isScreenshotEnabled()) {
             eventListener  = new TestEventForScreenshotListener(driver);
         }
-        enableScreenshots();
+        driver = enableScreenshots();
         eventListener.updateTestID(testName);
     }
 
     private WebDriver enableScreenshots() {
         log.debug("Enabling screenshot module...");
-        return EventFiringWebDriver.class.cast(driver).register(eventListener);
+        return new EventFiringWebDriver(driver).register(eventListener);
```

Comment:
Could we just do this in `createDriver()` ?  The fact that `enableScreenshots()` returns a `WebDriver` (which is really an `EventFiringWebDriver`) rather than `void` is all a bit non-obvious.